lib/zspec/queue.rb

```diff
@@ -96,6 +106,12 @@ def retry_count(message)
       @sink.hget(@metadata_hash_name, retry_key(message)).to_i
     end
 
+    def track_failure(message)
+      failure = JSON.parse(@sink.hget(@failure_hash_name, message) || "{count: 0}")
+      failure.merge({message: message, last_failure: @sink.time, count: failure[:count].to_i+1 })
```

Comment:
```suggestion
      failure.merge!({message: message, last_failure: @sink.time, count: failure[:count].to_i+1 })
```
Need `merge!` or `failure` will not update.